src/NuGet.Core/NuGet.Configuration/Settings/SettingsFile.cs

```diff
@@ -108,15 +114,50 @@ public SettingsFile(string directoryPath, string fileName, bool isMachineWide, b
             IsMachineWide = isMachineWide;
             IsReadOnly = IsMachineWide || isReadOnly;
 
-            XDocument config = null;
-            ExecuteSynchronized(() =>
+            NuGetEventSource.Instance.Write(
+                ConfigurationSettingsFileRead,
+                new EventSourceOptions
+                {
+                    ActivityOptions = EventActivityOptions.Detachable,
+                    Keywords = NuGetEventSource.Keywords.Configuration,
+                    Opcode = EventOpcode.Start,
+                },
+                new
+                {
+                    ConfigFilePath,
```

Comment:
anonymous types are classes, and therefore allocated on the heap.  Should we use `EventSource.IsEnabled(EventLevel, EventKeywords)` to avoid the allocation when nothing is listening to the event?

Same question applies to all the other usage of anonymous types for EventSource events, but this one, reading nuget.config files, is the one that will run most frequently.

Since `EventSource.Write` with a `data` overload all use generics, it means that if you define a struct for the data, then it should be possible to write the EventSource event with zero allocations. But it requires a bit more boilerplate, defining a struct for the data.